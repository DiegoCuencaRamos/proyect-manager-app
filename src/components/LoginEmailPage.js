import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { startEmailLogin, startEmailSignUp } from '../actions/auth'

const LoginEmailPage = () => {
    // 1. State
    const [ email, setEmail ] = useState('')
    const [ password, setPassword ] = useState('')
    const [ errorMessage, setErrorMessage ] = useState('')
    // 2. Variables
    const dispatch = useDispatch()
    const emailIsValid = (email) => {
        return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
    }

    // 3. vents
    const onEmailChange = (e) => {
        const value = e.target.value
        setEmail(value)
    }

    const onPasswordChange = (e) => {
        const value = e.target.value
        setPassword(value)
    }

    const onEmailLogin = (e) => {
        e.preventDefault()
        dispatch(startEmailLogin(email, password, setErrorMessage))
    }

    const onEmailSingUp = (e) => {
        e.preventDefault()
        setErrorMessage('')
        const emailValid = emailIsValid(email)
        
        if (emailValid && password) {
            dispatch(startEmailSignUp(email, password, setErrorMessage))
        } else {
            setErrorMessage('Please provid a valid email and password')
        }
    }

    // 4. Render
    return (
        <section className="email-login-page">
            <div className="container">
                <h1 className="email-login-page__title">Login with your email</h1>
                <h5>Please provide email and password</h5>
                {errorMessage && <p className="error-message">{errorMessage}</p>}
                <form className="email-form__wrapper">
                    <input
                        id="txtEmail" 
                        className="form__item--emial"
                        type="text"
                        value={email}
                        placeholder="Email"
                        onChange={onEmailChange}
                    />
                    <input 
                        id="txtPassword"
                        className="form__item--password"
                        type="password"
                        value={password}
                        placeholder="Password"
                        onChange={onPasswordChange}
                    />
                    <div className="form__buttom-wrpper">
                        <button 
                            id={'emailLoginBtn'}
                            className="button"
                            onClick={onEmailLogin}
                        >Login</button>
                        <button 
                            id={'emailSignUpBtn'}
                            className="button--remove"
                            onClick={onEmailSingUp}
                        >Sign Up</button>
                    </div>
                </form>
            </div>
        </section>
    )
}

export default LoginEmailPage