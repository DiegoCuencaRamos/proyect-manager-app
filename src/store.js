import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store