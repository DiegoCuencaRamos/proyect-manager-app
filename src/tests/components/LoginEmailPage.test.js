import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import store from '../../store'
import LoginEmailPage from '../../components/LoginEmailPage'

let wrapper

beforeEach(() => {
    wrapper = mount(
        <Provider store={store}>
            <LoginEmailPage />
        </Provider>
    )
})

test('Should render LoginPage correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should set email on email change', () => {
    const value = 'any email'
    wrapper.find('.form__item--emial').simulate('change', {
        target: { value }
    })
    expect(wrapper.find('.form__item--emial').prop('value')).toBe(value)
})

test('Should set password on password change', () => {
    const value = 'any password'
    wrapper.find('.form__item--password').simulate('change', {
        target: { value }
    })
    expect(wrapper.find('.form__item--password').prop('value')).toBe(value)
})

// Falta testear events: Redux - dispatchs.
