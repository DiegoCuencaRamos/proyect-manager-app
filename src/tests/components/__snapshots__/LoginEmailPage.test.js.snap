// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <LoginEmailPage>
    <section
      className="email-login-page"
    >
      <div
        className="container"
      >
        <h1
          className="email-login-page__title"
        >
          Login with your email
        </h1>
        <h5>
          Please provide email and password
        </h5>
        <form
          className="email-form__wrapper"
        >
          <input
            className="form__item--emial"
            id="txtEmail"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <input
            className="form__item--password"
            id="txtPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <div
            className="form__buttom-wrpper"
          >
            <button
              className="button"
              id="emailLoginBtn"
              onClick={[Function]}
            >
              Login
            </button>
            <button
              className="button--remove"
              id="emailSignUpBtn"
              onClick={[Function]}
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
    </section>
  </LoginEmailPage>
</Provider>
`;
